entity/templates/src/main/webapp/app/_entity-controller.js

```diff
@@ -45,23 +45,25 @@ angular.module('<%=angularAppName%>')
 
         $scope.delete = function (id) {
             <%= entityClass %>.get({id: id}, function(result) {
-                $scope.<%= entityInstance %> = result;
-                $('#delete<%= entityClass %>Confirmation').modal('show');
-            });
-        };
 
-        $scope.confirmDelete = function (id) {
-            <%= entityClass %>.delete({id: id},
-                function () {
-                    <%_ if (pagination != 'infinite-scroll') { _%>
-                    $scope.loadAll();
-                    <%_ } else { _%>
-                    $scope.reset();
-                    <%_ } _%>
-                    $('#delete<%= entityClass %>Confirmation').modal('hide');
-                    $scope.clear();
+                var modalInstance = $modal.open({
```

Comment:
If you think having state is ok ill rewrite it as it is
On 1 Nov 2015 17:07, "Zsombor" <notifications@github.com> wrote:

> In entity/templates/src/main/webapp/app/_entity-controller.js
> <https://github.com/jhipster/generator-jhipster/pull/2255#discussion_r43580364>
> :
>
> >
> > -        $scope.confirmDelete = function (id) {
> > -            <%= entityClass %>.delete({id: id},
> > -                function () {
> > -                    <%_ if (pagination != 'infinite-scroll') { _%>
> > -                    $scope.loadAll();
> > -                    <%_ } else { _%>
> > -                    $scope.reset();
> > -                    <%_ } _%>
> > -                    $('#delete<%= entityClass %>Confirmation').modal('hide');
> > -                    $scope.clear();
> > +                var modalInstance = $modal.open({
>
> I prefer to have as many separate state, as possible, so it is easier to
> re-use the same functionality in multiple places. Could you put this code
> into a separate service? So it can be called from the details page, as well
> as from the list page.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jhipster/generator-jhipster/pull/2255/files#r43580364>
> .
>